[Agent Management] Add Handling for HTTP 304 Not Modified Response (#5260)
grafana/agent: docs/sources/static/configuration/agent-management.md

    # Whether to use labels from the label management service. If enabled, labels from the API supersede the ones configured in the agent.
    label_management_enabled: <bool> | default = false

>     # Whether to accept HTTP Not Modified (304) responses from the API server. If enabled, the agent will use the cached configuration if the API server responds with a 304. This can be disabled for debugging or testing, but will save network bandwidth and CPU time if enabled.

Reviewer: ```suggestion
    # Whether to accept HTTP 304 Not Modified responses from the API server. If enabled, the agent will use the cached configuration if the API server responds with HTTP 304 Not Modified. You can set this argument to `false` for debugging or testing.
```

I'm not sure that we need to state `Using the cached configuration saves network bandwidth and CPU time. ` This is implied by the HTTP 304 and redirecting to the cache. Also this statement isn't quantified in any way here.

>     accept_http_not_modified: <bool> | default = false

Reviewer 2: any reason not to make this default to `true`?

Reviewer: I was going to ask the same. A default to `false` tells me that if a 304 is received, requested resources are sent again instead of redirecting to the cached resources. That seems backwards to me.
